src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,10 +27,23 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+        String[] splitAddress = trimmedAddress.split(",");
+        int numParts = splitAddress.length;
+
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        this.block = new Block(splitAddress[0]);
+        if (numParts > 1) {
+            this.street = new Street(splitAddress[1]);
+        }
+        if(numParts > 2) {
+            this.unit = new Unit(splitAddress[2]);
+        }
+        if (numParts > 3) {
```

Comment:
Can use a better variable name than 'numParts' to make it more readable. Also avoid "magic numbers" like 1,2 etc.